packages/next/src/server/image-optimizer.ts

```diff
@@ -569,13 +569,24 @@ export async function imageOptimizer(
         )
       }
 
+      if (mocked.res.statusCode === 404) {
```

Comment:
I was thinking that it might be better to throw an ImageError right here for status codes other than 200-299, similar to what is done when isAbsolute=true. 
However, for now, I have limited the implementation to handle only 404 errors.

Reply:
yeah perhaps an `ok` property can be added to `mock-request.ts` response. That way it's a similar implementation how the other image check is working for `isAbsolute` images